cmd/commands.go

```diff
@@ -63,16 +63,25 @@ func Migrate(c config.Reader, tenant string, listTenantsFlag bool) error {
 
 	db = db.WithAutomigrate().(*mongo.DataStoreMongo)
 
-	tenantCtx := identity.WithContext(context.Background(), &identity.Identity{
-		Tenant: tenant,
-	})
-
-	dbname := mstore.DbFromContext(tenantCtx, mongo.DbName)
-	if err != nil {
-		return errors.Wrap(err, "failed to decode dbname")
+	if config.Config.Get(dconfig.SettingTenantAdmAddr) != "" {
+		db = db.WithMultitenant()
 	}
 
-	err = db.MigrateTenant(tenantCtx, dbname, mongo.DbVersion)
+	ctx := context.Background()
+	if tenant == "" {
+		err = db.Migrate(ctx, mongo.DbVersion)
+	} else {
+		tenantCtx := identity.WithContext(ctx, &identity.Identity{
+			Tenant: tenant,
+		})
+
+		dbname := mstore.DbFromContext(tenantCtx, mongo.DbName)
+		if err != nil {
```

Comment:
where does `err` come from in here? `DbFromContext` returns string, we could check if it is `""` but looking at the code it could only happen if mongo.DbName is "".

Reply:
uh, interesting, you are right... this comes from the code already there (I just made it conditional with an if). Fixed.